src/extension.ts

```diff
@@ -21,29 +22,442 @@ import CMakeTools from './cmake-tools';
 import rollbar from './rollbar';
 import {DirectoryContext} from '@cmt/workspace';
 import {StateManager} from '@cmt/state';
+import {Kit, readKitsFile, scanForKits, descriptionForKit, USER_KITS_FILEPATH} from '@cmt/kit';
+import {fs} from '@cmt/pr';
+import {MultiWatcher} from '@cmt/watcher';
 
-let INSTANCE: CMakeTools|null = null;
+/**
+ * A class to manage the extension.
+ *
+ * Yeah, yeah. It's another "Manager", but this is to be the only one.
+ *
+ * This is the true "singleton" of the extension. It acts as the glue between
+ * the lower layers and the VSCode UX. When a user presses a button to
+ * necessitate user input, this class acts as intermediary and will send
+ * important information down to the lower layers.
+ */
+class ExtensionManager implements vscode.Disposable {
```

Comment:
I think this class is responsible for three issues.

-  Workspace coordinator
   It monitors changes on the workspace and forwards commands to the workspace relatee cmake instances. 

- cmaketool instances management
 Each workspace has a direct relation to one cmaketool instance. This means the Information and objects are tidy coupled to each other. I would separate them into a class which is used by the workspacecoordinator (ExtrnsionManager) 

- Kit file management
  The kit file management is a big topic and it has two levels user related kits and workspace related kits. The scan part and the collection of user kits is a user related task and is independent from the workspace. I would separate it from the workspace coordination. The workspace use only user kits by the select kit commands. But the workspace oder cmaketools instance need to manage the actual kit. I would forward the kit selection to the cmaketools. this would reduce the responsabilities for the extension manager.